conjure-undertow-lib/src/main/java/com/palantir/conjure/java/undertow/lib/WrappedEndpoint.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * (c) Copyright 2019 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.conjure.java.undertow.lib;
+
+import io.undertow.server.HandlerWrapper;
+import io.undertow.server.HttpHandler;
+import java.util.function.BiFunction;
+
+/**
+ * Wraps all the {@link HttpHandler} of an {@link Endpoint} with an HandlerWrapper defined as a function
+ * that takes an {@link EndpointDetails} and a {@link HttpHandler} and return the wrapped (or not) {@link HttpHandler}.
+ * Useful when you need to add a custom wrapper to an individual endpoint.
+ */
+public final class WrappedEndpoint implements Endpoint {
```

Comment:
I'd prefer to hold off making this sort of utility public until the system is a bit more stable

Reply:
If we don't make this public, then there isn't much sense to this PR since it's not used internally either. Would it be enough to make it `@Beta` ?